crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -2379,6 +2380,44 @@ impl<TX: DbTxMut + DbTx + 'static, N: NodeTypes> TrieWriter for DatabaseProvider
 
         Ok(num_entries)
     }
+
+    fn clear_trie_changesets(&self) -> ProviderResult<()> {
+        let tx = self.tx_ref();
+        tx.clear::<tables::AccountsTrieChangeSets>()?;
+        tx.clear::<tables::StoragesTrieChangeSets>()?;
+        Ok(())
+    }
+
+    fn clear_trie_changesets_range(
+        &self,
+        range: impl RangeBounds<BlockNumber>,
```

Comment:
something we might want to consider immediately now is that if we move trie changesets to static files, we would not be able to provide a `range`, it will just be "truncate N from the end". Maybe we should change it to be a truncate now?

Reply:
Changed to `clear_trie_changesets_from`.

I'm not sure it's going to make sense to have these in static files tho, at least in their current state. Without an accompanying index the data isn't very useful beyond reverting a relatively small amount of blocks, so storing in static files would have us keep way more than we could use for no benefit.

Reply:
yeah, the indices would live in the DB same as for account/storage history, but the changesets themselves in static files. This can improve storage and perf of querying actual trie changesets.